### PR TITLE
nwg-drawer: 0.1.11 -> 0.2.8

### PR DESCRIPTION
**Pull request by sei40kr**
###### Motivation for this change

Update [nwg-drawer](https://github.com/nwg-piotr/nwg-drawer) 0.1.11 -> ~0.2.6~ 0.2.7.

Also added a patch to refer resources under `$out/share`.
It can be changed by setting `XDG_DATA_HOME` but changing it is too risky so I ended up to create a patch.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by plabadens — APPROVED** [review]
These changes look good.

**Comment by c0bw3b** [comment]
@sei40kr upstream has a 0.2.8 version which includes a fix for high CPU load.
Could you refresh your PR to update to this version please?

**Comment by sei40kr** [comment]
@c0bw3b 
Just updated to 0.2.8.

But some icons were broken for some reason.
I checked commit logs between 0.2.7 -> 0.2.8 but it didn't touch the icons.

I'm still investigating. Let me make this PR a draft.

**Comment by sei40kr** [comment]
@c0bw3b 
> But some icons were broken for some reason.

Sorry I could reproduce this in the older version of nwg-drawer.
And it worked after restarting sway session. I don't still know why. But it's no need to fix it in this PR, I think.

Could you review again?

File: pkgs/applications/misc/nwg-drawer/default.nix [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
-  nativeBuildInputs = [ pkg-config ];
+  preInstall = ''
+    mkdir -p $out/share/nwg-drawer
+    cp -r desktop-directories drawer.css $out/share/nwg-drawer
+  '';
 
-  buildInputs = [ cairo gobject-introspection gtk3 gtk-layer-shell ];
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix PATH : ${xdg-utils}/bin
```
Comment (sei40kr) [inline pkgs/applications/misc/nwg-drawer/default.nix:37]: nwg-drawer needs `xdg-utils`, so I added it.

**Review by c0bw3b — APPROVED** [review]
Result of `nixpkgs-review pr 155123` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nwg-drawer</li>
  </ul>
</details>

Works as expected.

I'll wait for ofborg builds to finish before merging.

**Pull request merged**